content/blog/2022-04-26-e2e-cv-pt1.md

```diff
@@ -0,0 +1,331 @@
+time. All other stages have this text file as a dependency. Thus, the entire
+pipeline will be rerun if anything about our python environment changes.
+
+We can see the whole dependency graph (directed acyclic graph, to be exact)
+using the [`dvc dag`](https://dvc.org/doc/command-reference/dag) command:
+
+```bash
+$ dvc dag
+                               +----------------+
+                               | check_packages |
+                          *****+----------------+
+                    *****         *    **    **
+               ****             **       **     ***
+            ***               **            **      ***
++-----------+               **               *         ***
+| data_load |             **                 *             *
++-----------+           **                   *             *
+           ***        **                     *             *
+              *     **                       *             *
+               **  *                         *             *
+          +------------+                     *             *
+          | data_split |***                  *             *
+          +------------+   ***               *             *
+                  *            ***           *             *
+                  *               ***        *             *
+                  *                   **     *             *
+                  **                    +-------+        ***
+                    ***                 | train |     ***
+                        ***             +-------+   ***
+                            ***        **        ***
+                               ***   **      ***
+                                   **    ***
+                                  +----------+
+                                  | evaluate |
+                                  +----------+
+```
+
+The entire pipeline can be easily reproduced with the `dvc exp run` command:
+
+```bash
+$ dvc exp run
+Running stage 'check_packages':
+> python src/stages/check_pkgs.py --config=params.yaml
+...
+Running stage 'data_load':
+> python src/stages/data_load.py --config=params.yaml
+...
+Running stage 'data_split':
+> python src/stages/data_split.py --config=params.yaml
+...
+Running stage 'train':
+> python src/stages/train.py --config=params.yaml
+...
+Running stage 'evaluate':
+> python src/stages/eval.py --config=params.yaml
+...
+```
+
+Our DVC pipeline relies on the parameters defined in the
+[`params.yaml`](https://github.com/iterative/magnetic-tiles-defect/blob/main/params.yaml)
+file. By loading its contents in each stage, we can avoid hard-coded parameters.
+It also allows rerunning the whole or parts of our pipeline under a different
+set of parameters. The DVC pipeline YAML file `dvc.yaml` supports a
+[templating format](https://dvc.org/doc/user-guide/project-structure/pipelines-files#templating)
+to insert values from different sources in the YAML structure itself.
```

Comment:
It seems like this is more relevant to part 2 of the blog post? I think you could probably skip it here.